### PR TITLE
Always reauthenticate before password change

### PR DESCRIPTION
**Pull request by whaught**
Issue https://github.com/google/exposure-notifications-verification-server/issues/1168

## Proposed Changes

* Share the authentication code with login
* Always have the user reauthenticate before changing password
* This eliminates issues with Firebase login-recency requirement and is more secure (eg. shared machines).
* Start moving some out stuff to application.js where it can be better cached

### CODE REVIEW

File: cmd/server/assets/login/_loginscripts.html [hunk 1]
```diff
@@ -155,67 +155,289 @@
...
       $retyped.addClass("text-muted");
+      $retyped.removeClass("text-danger");
     }
 
     {{end}}
     return valid;
   }
 {{end}}
+
+{{define "login/login-js"}}
```
Comment (sethvargo) [inline cmd/server/assets/login/_loginscripts.html:235]: Is it feasible to move this to application.js while we're moving things around? Anything that depends on server-side information we can populate in the HTML with hidden fields and then pull them in the js.

**Review by sethvargo — COMMENTED** [review]
/lgtm
/hold 
for one comment

File: cmd/server/assets/login/_loginscripts.html (hunk 1 above)
Comment (whaught) [reply cmd/server/assets/login/_loginscripts.html:134]: move password complexity into application.js and moved server-side stuff into a js struct to use as a param.

still looking at the rest, but I get a little concerned by the complexity of this refactor

**Review by sethvargo — APPROVED** [review]
/lgtm

**Pull request merged**